pmaudit

```diff
@@ -167,16 +167,33 @@ def os_path_walk(top, func, arg):
         if stat.S_ISDIR(st.st_mode):
             os_path_walk(name, func, arg)
 
+def die(message):
+    """Report failure message and exit program with failure"""
+    print(message, file=sys.stderr)
+    sys.stderr.flush()
+    exit(1)
+
+def mkdir_p_or_die(new_dir):
+    """Make a directory or die"""
+    if not new_dir or new_dir == '.':
+        return # Presume empty/current directory exists.
+    try:
```

Comment:
My version used a try/raise to avoid a race condition but I don't see any point in using it along with exists_ok (which very likely does the same basic thing internally). I believe in letting exceptions flow when it's a real error. Exception backtraces are not the Python equivalent of core dumps in C, something to be avoided wherever possible; they're just the default Python error message format and should not IMHO be overridden without a good reason. In this case, if we can't create a directory the best thing to do is let the exception tell us why and where. The code becomes simpler and the failure message is more descriptive - what's not to like about that?

Reply:
I think letting expected exceptions flow out to the user is user-hostile, because they are *exactly* the same as C core dumps; good for debugging, but not something users should have to see in normal operation.  But it's your program, I'll do it.

It's late where I am, I'll have to work on these comments later.

Reply:
Exception handler removed.